### ISSUE TITLE
PS2EXE may be sucking up the compiled scripts

### ISSUE THREAD

**Issue by GarbetFibornil** [MScholtes/PS2EXE#127]
Hello,
Yesterday a colleague used your executable to compile a posh script that was unfortunately sending mails through an authenticated mail relay, and hard-coded the password in it (I know, not good practice, and know we have changed the password).
Very strangely, he did NOT distribute the compiled exe whatsoever, but only a few hours later, we received three emails obviously sent from the script.
I therefore believe that your executable is **uploading somewhere any script compiled with it**. Which is not good !

I am NO expert at all neither in code nor in github, but I noticed that there are two repos, one is https://github.com/MScholtes/Win-PS2EXE where I can find source code (and, being no C# specialist, I looked at it and could not see any evidence that you would be uploading the script) and this repo where, if I am right, we can't find the source code.

Please explain ?

**Comment by MScholtes** [comment]
I have rarely heard such a cheeky statement: you are careless with passwords and then you suspect someone else of being responsible when they are misused!

The source code of PS2EXE and also Win-PS2EXE is public and it is verifiable that it does not contain any malicious code. This tool has been in use for a long time and I have never heard a complaint of this kind.

Please urgently check your work processes and stop blaming others for your mistakes.

**Just imagine the damage you would do to this project if anyone believed you!**

**Comment by GarbetFibornil** [comment]
Hi @MScholtes ,
thanks for your quick answer. I did not mean to blame anyone, and I DID mention that what we did was NOT good practice.

I am just stating the facts that we observed : The script that we compiled with the downloaded executable was run by people we don't know, and we did not share at all the script or the executable. Basically the script is simply collecting some information on the computer / user, for inventory purposes. Nothing really sensitive : cpu type, amount of ram, stuff like that.

Also, as mentioned, I did check the source code and found nothing that could explain this. But I am puzzled by the fact that you have two repos (ps2exe, where I can find the source), and this one, where I can only find the compiled exe.

If it is of any interest to you, I am ok to share (but probably privately rather than here) a screenshot of three emails that we received, that were the output of the script.

**Comment by MScholtes** [comment]
I still find it unbelievable that you are trying to find people to blame here for your mistakes:

1. On the readme webpage it is pointed out to never store passwords in scripts you compile with PS2EXE: https://github.com/MScholtes/PS2EXE/tree/master#password-security. What you have done is not "not good practice", it is an invitation to hackers or too curious colleagues.
2. The source code of both tools is in the repository: the one for Win-PS2EXE is surprisingly in the subfolder Win-PS2EXE: https://github.com/MScholtes/PS2EXE/tree/master/Win-PS2EXE. PS2EXE itself is a Powershell script, this is delivered as a readable script (the source is here: https://github.com/MScholtes/PS2EXE/blob/master/Module/ps2exe.ps1).
3. You have not mentioned any evidence that PS2EXE could be to blame for stealing your script/program. Your reasoning is: It can't be me ("it's me, so innocent"), so it must be someone else!

Please tell "your colleague" to be careful with company secrets in future!
I'm not interested in screenshots or anything like that, just be careful with your work, please.

**Comment by GarbetFibornil** [comment]
Good afternoon,

In the mean time, I did my homework, and can confirm that :
1/ It's pure coincidence and It absolutely has **nothing** to do with Win-PS2EXE, and
2/ I can reproduce it at will.

What happens is that our antivirus has sandboxing functionality. When Win-PS2EXE opens the script, the antivirus uploads the script and runs it in the sandbox.

However you kept missing my point in your answers. I never meant to blame anyone. This script with a hard-coded password was a bad idea, I know. But it's exactly because this script is sending emails that we got the result we got. So you're focusing on our bad practice, while I'm simply mentioning that for some reason my script gets "caught" somewhere and executed.

Indeed, my first idea was that the downloadable executable itself had been hacked -I guess this always remains possible.

Reading back my first post, I do realise that it was probably aggressive in the way it was written, and I am sorry for that. But I also think your answers were quite violent, and as already said, off-topic.
